packages/adapter/src/lib/adapter/defaultObjs.ts

```diff
@@ -1,4 +1,4 @@
-function createDefaults(): Record<string, Partial<ioBroker.StateCommon>> {
+export function createDefaults(): Record<string, Partial<ioBroker.StateCommon>> {
```

Comment:
Do we want the returned object to automatically know its keys or is `string` good enough? Maybe it is even desirable if we are working with unknown `string` keys anyways.

Reply:
I have created an id type in https://github.com/ioBroker/ioBroker.js-controller/blob/d6867b6e69676bc0054d0b6b4fa8deb7d88a4030/packages/adapter/src/lib/adapter/utils.ts#L8 

It is not yet used consistently. I tend to do such optimizations which I see as really optional after this PR. Or do you mean something else here?

Reply:
I meant instead of returning a `Record<string, ...>` where the caller doesn't know which keys exist, we can have this inferred automatically. It requires a bit of type magic though.

But maybe this is undesirable, since there might be errors like `string isn't assignable to "foo" | "bar" | ...`.